graphdatascience/model/simple_edge_embedding_model.py

```diff
@@ -0,0 +1,147 @@
+from typing import Any, Dict, List, Union
+
+from pandas import DataFrame, Series
+
+from ..query_runner.query_runner import QueryRunner
+from ..server_version.server_version import ServerVersion
+
+
+class SimpleEdgeEmbeddingModel:
+    def __init__(
+        self,
+        scoring_function: str,
+        query_runner: QueryRunner,
+        server_version: ServerVersion,
+        graph_name: str,
+        node_embedding_property: str,
+        relationship_type_embeddings: Dict[str, List[float]],
+    ):
+        self._scoring_function = scoring_function
+        self._query_runner = query_runner
+        self._server_version = server_version
+        self._graph_name = graph_name
+        self._node_embedding_property = node_embedding_property
+        self._relationship_type_embeddings = relationship_type_embeddings
+
+    def predict_stream(
+        self,
+        source_node_spec: Union[str, List[int]],
+        target_node_spec: Union[str, List[int]],
```

Comment:
couldnt the spec also be a list of labels?

Reply:
NIT - could define a type for this Union

Reply:
I guess so, but that was not what was specified in the scoping doc I think

Reply:
Added a type for it. And single node IDs are also allowed, according to the server implementation